reconcile/utils/state.py

```diff
@@ -53,14 +53,26 @@ def ls(self):
         """
         Returns a list of keys in the state
         """
-        objects = self.client.list_objects(Bucket=self.bucket,
-                                           Prefix=f'{self.state_path}/')
+        objects = self.client.list_objects_v2(Bucket=self.bucket,
+                                              Prefix=f'{self.state_path}/')
 
         if 'Contents' not in objects:
             return []
 
-        return [o['Key'].replace(self.state_path, '')
-                for o in objects['Contents']]
+        response = [o['Key'].replace(self.state_path, '')
+                    for o in objects['Contents']]
+
+        while objects['IsTruncated']:
+            objects = self.client.list_objects_v2(
+                Bucket=self.bucket,
+                Prefix=f'{self.state_path}/',
+                ContinuationToken=objects['NextContinuationToken']
+            )
+
+            response += [o['Key'].replace(self.state_path, '')
```

Comment:
Why not do this replacement only once and at the very end of the method?

Reply:
Done. And thanks.